plugin/fulltext/ngram_parser/plugin_ngram.cc

```diff
@@ -0,0 +1,238 @@
+/*
+   Copyright (c) 2014, 2017, Oracle and/or its affiliates. All rights reserved.
+   Copyright (c) 2020, MariaDB.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License, version 2.0,
+   as published by the Free Software Foundation.
+
+   This program is also distributed with certain software (including
+   but not limited to OpenSSL) that is licensed under separate terms,
+   as designated in a particular file or component or in included license
+   documentation.  The authors of MySQL hereby grant you an additional
+   permission to link the program and your derivative works with the
+   separately licensed software that they have included with MySQL.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License, version 2.0, for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#include <stddef.h>
+#include <limits.h>
+#include <my_attribute.h>
+#include <my_global.h>
+#include "../storage/innobase/include/fts0tokenize.h"
```

Comment:
Could we move the needed definitions from `fts0tokenize.h` to a ‘global’ file and `#include` it from both files?

Reply:
As were requested on a side channel, I rebased code onto 10.2 branch. There are already changes in functions (my_ci_charlen, for example) between major branches, that will make merging code into other branches cumbersome. Perhaps, it's better to leave that code move for later?

If fulltext search engine is going to be extracted from storage engines code for consolidation, that move will be performed anyway. I think it's probably fine to leave it where it's now at the moment.

Reply:
I think that @vuvova should comment on this.